src/main/scala/com/microsoft/hyperspace/actions/Action.scala

```diff
@@ -80,17 +81,24 @@ trait Action extends Logging {
   }
 
   def run(): Unit = {
+    val appInfo =
+      AppInfo(sparkContext.sparkUser, sparkContext.applicationId, sparkContext.appName)
```

Comment:
I have seen it have issues with unit tests, e.g. in JoinRUle and FilterRule unit tests, where once this val is fixed, it somehow refers to incorrect spark session leading to test failures. (This was seen when individually tests would pass, but the test suite when run as a whole would fail)

Reply:
I think we need to understand why (something doesn't seem to be set up correct). Can you create an issue after this PR is merged?

Reply:
got it. I will add an issue